ui/lib/kv/addon/components/page/list.js

```diff
@@ -15,33 +15,45 @@ import errorMessage from 'vault/utils/error-message';
  * @module List
  * ListPage component is a component to show a list of kv/metadata secrets.
  *
- * @param {array} model - An array of models generated form kv/metadata query.
- * @param {array} breadcrumbs - Breadcrumbs as an array of objects that contain label, route, and modelId. They are updated via the util kv-breadcrumbs to handle dynamic *pathToSecret on the list-directory route.
+ * @param {array} secrets - An array of models generated form kv/metadata query.
+ * @param {string} backend - The name of the kv secret engine.
+ * @param {string} pathToSecret - The directory name that the secret belongs to ex: beep/boop/
+ * @param {string} pageFilter - The input on the kv-list-filter. Does not include a directory name.
+ * @param {string} filterValue - The concatenation of the pathToSecret and pageFilter ex: beep/boop/my-
  * @param {boolean} noMetadataListPermissions - true if the return to query metadata LIST is 403, indicating the user does not have permissions to that endpoint.
+ * @param {array} breadcrumbs - Breadcrumbs as an array of objects that contain label, route, and modelId. They are updated via the util kv-breadcrumbs to handle dynamic *pathToSecret on the list-directory route.
+ * @param {string} routeName - Either list or list-directory.
+ * @param {object} meta - Object with values needed for pagination, created by LazyPaginatedQuery on the store service.
  */
 
 export default class KvListPageComponent extends Component {
   @service flashMessages;
   @service router;
+  @service store;
 
   @tracked secretPath = '';
 
   get mountPoint() {
-    // mountPoint tells the LinkedBlock component where to start the transition. In this case, mountPoint will always be vault.cluster.secrets.backend.kv.
+    // mountPoint tells transition where to start. In this case, mountPoint will always be vault.cluster.secrets.backend.kv.
     return getOwner(this).mountPoint;
   }
 
   @action
   async onDelete(model) {
     try {
-      const message = `Successfully deleted the metadata and all version data of the secret ${model.fullSecretPath}.`;
+      // The model passed in is a kv/metadata model
```

Comment:
It looks like `secret` is passed in the hbs template which I thought was a `kv/data` model?

Reply:
Words are difficult for this. So it's the secret returned on the kv/metadata query LIST. This is why I thought it was the wrong model for the destroyRecord. But because its coming from the kv/metadata adapter, it's returning the kv/metadata model. 

I could rename this so it's not so confusing. 

Any thoughts?

Reply:
I see - this is the confirm action from the list view - thank you for clarifying! I think the comment here is sufficient plus renaming the template from `secret` to metadata makes this clearer!